scripts/globals/mobskills/soulshattering_roar.lua

```diff
@@ -11,5 +11,17 @@ function onMobSkillCheck(target,mob,skill)
 end
 
 function onMobWeaponSkill(target, mob, skill)
-    return 0
+    local typeEffect = tpz.effect.TERROR
+    local duration = 10
+    local dmgmod = 2.0
+
+    MobPhysicalStatusEffectMove(mob, target, skill, typeEffect, 1, 0, 30)
+
+    local info = MobMagicalMove(mob, target, skill, mob:getWeaponDmg() * 4, tpz.magic.ele.DARK, dmgmod,TP_MAB_BONUS, 1)
+    local dmg = MobFinalAdjustments(info.dmg, mob, skill, target, tpz.attackType.MAGICAL, tpz.damageType.DARK, MOBPARAM_WIPE_SHADOWS)
+    target:takeDamage(dmg, mob, tpz.attackType.MAGICAL, tpz.damageType.DARK)
+
+    -- TODO: Temporary immunity to a single weapon damage type
```

Comment:
After merge someone will have to file an issue so we don't forget that this aspect isn't implemented.

We do have damage type immunities, but those are "set and forget" in nature, and the methods involved with making them temporary would be ugly. Will need dedicated effects for them in the future so limited duration can be baked into the initial call.